### ISSUE TITLE
Unable run train if in system was disabled IPV6 thru kernel params.

### ISSUE THREAD

**Issue by james007blond** [anhvth/opensloth#14]
My introduction to "multigpu" testing started from offered [Kaggle's Qwen3-Opensloth-2GPUS](https://www.kaggle.com/code/anhvth226/qwen3-opensloth-2gpus)

Form a "door" i've got this errors

```
[W729 12:37:15.952489580 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [localhost]:29501 (errno: 97 - Address family not supported by protocol).
[W729 12:37:15.041435752 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [localhost]:29501 (errno: 97 - Address family not supported by protocol).
[W729 12:37:15.044199686 CUDAAllocatorConfig.h:28] Warning: expandable_segments not supported on this platform (function operator())
[W729 12:37:15.046431838 CUDAAllocatorConfig.h:28] Warning: expandable_segments not supported on this platform (function operator())

```

And of course tried solve issue by adding some debugging in script beggining

```
test:test~$ cat train.py
import torch
import os
import socket

os.environ['NCCL_SOCKET_FAMILY'] = 'AF_INET'     # Strict use IPV4 NCCL
os.environ['MASTER_ADDR'] = '192.168.0.100'        # Address
os.environ['NCCL_SOCKET_IFNAME'] = 'eth0'          # Interface
os.environ['MASTER_PORT'] = '29500'                      # Port


print(torch.cuda.is_available())
print(torch.version.cuda)
print(os.environ.get('MASTER_ADDR'))
print(os.environ.get('MASTER_PORT'))
print(os.environ.get('NCCL_SOCKET_IFNAME'))

if torch.cuda.is_available():
    backend = 'nccl'
else:
    backend = 'gloo'

torch.distributed.init_process_group(backend=backend 

```

```
(env) test@test:~/notebook$ python train.py
True
12.4
[W729 14:26:19.507906024 socket.cpp:759] [c10d] The client socket cannot be initialized to connect to [::ffff:192.168.0.100]:29500 (errno: 97 - Address family not supported by protocol).
```

```
test:test~$ cat /etc/default/grub
GRUB_CMDLINE_LINUX_DEFAULT="ipv6.disable=1 ..........................."
```

If enable IPV6 in kernel - all goes as should. I didn't find popper OS.ENV param and this issue didn't direct related to opensloth, but now it uses NVCCL via sockets and not via MMGS.

May be your team implement "auto switch" between Memory-mapped gradient synchronization (MMGS) и NCCL-based gradient synchronization.
